File: java/singleton/singleton/src/main/java/edu/uea/Singleton.java
package edu.uea;

public class Singleton {
	// Instância privada e estática da própria classe
	private static Singleton instance;

	// Construtor privado para impedir a criação de novos objetos fora da classe
	private Singleton() {
	}

	// Método público e estático para obter a instância única
	public static synchronized Singleton getInstance() {
		if (instance == null) {
			// Cria a instância caso ainda não exista
			instance = new Singleton();
		}
		return instance;
	}

	// Exemplo de método que pode ser acessado pela instância Singleton
	public void showMessage(String mensage) {
		System.out.println(mensage);
	}

}

File: python/singleton/exemplo1/Singleton.py
class Singleton:
    # Atributo de classe que armazenará a instância única
    _instance = None

    # Método especial __new__ para controlar a criação da instância
    def __new__(cls):
        if cls._instance is None:
            # Cria a instância caso ainda não exista
            cls._instance = super(Singleton, cls).__new__(cls)
        return cls._instance

    # Exemplo de método que pode ser acessado pela instância Singleton
    def show_message(self, mesage):
        print(mesage)